optuna/_transform.py

```diff
@@ -46,6 +46,10 @@ class _SearchSpaceTransform:
     Attributes:
         bounds:
             Constructed bounds from the given search space.
+        column_to_encoded_columns:
+            Constructed mapping from original parameter column index to encoded column indices.
+        encoded_column_to_column:
+            Constructed mapping from encoded column index to original parameter column index.
```

Comment:
I can create a separate PR for the transform changes if you'd like. This PR might give some context.